### ISSUE TITLE
Problem with log rotation

### ISSUE THREAD

**Issue by prologic** [circuits/irclogger#5]
There seems to be an issue with log rotation with channels with _dashes_*.

**Comment by prologic** [comment]
Hmm I'm not seeing an issue with the code here (_het_).

We already log several channels with dashses in them:

http://irclogs.shortcircuit.net.au/%23crux-devel/

**Comment by jibanes** [comment]
It's noteworthy to comment that my "output" file resides on nfs.

**Comment by jibanes** [comment]
My command line:
export USER=mybot
irclogger -o /mypath/irc -c '#cat-v' irc.freenode.net

**Comment by prologic** [comment]
NFS eh? I can reproduce this at home then on my new shiny NAS :) I'll do some testing...

**Comment by prologic** [comment]
Thanks @jibanes -- Expect to see a duplicate logger join your chan :)

**Comment by jibanes** [comment]
Could it be a bug in circuits? I'm using 3.1.0
and python 2.7.3

**Comment by jibanes** [comment]
Oh sorry, I didn't know you wrote circuits too, nevermind that...

**Comment by jibanes** [comment]
modified filename (strftime("%Y-%m-%d-%H-%M-%S", localtime())))

and setting this in "init" doesn't rotate it either:

```
    #interval = datetime.fromordinal((
    #    date.today() + timedelta(1)
    #).toordinal())
    interval = 60
    Timer(interval, rotate(), self.channel).register(self)
```

**Comment by jibanes** [comment]
rotate does get called properly.

**Comment by jibanes** [comment]
I confirm the issue has nothing to do with presence of '-' in channel name.

**Comment by jibanes** [comment]
there's only one explanation, this is the issue:
        self.fire(open(path.join(dirname, logfile), "a"), self.channel)

**Comment by jibanes** [comment]
dirname and logfile are properly set.

**Comment by jibanes** [comment]
though this works:
        self.fire(close(), self.channel)

**Comment by jibanes** [comment]
this doesn't look right:
LOGFILE_REGEX = compile_regex("^(._).(._).log$")
note: apparently github doesn't display this right either

why do you have a dot between the captures?
ok I see why: logger.#channel

**Comment by jibanes** [comment]
is seems that generate_logfile returns something different from what parse_logfile provides.

**Comment by prologic** [comment]
I think I may have figured this out.

I've cleaned up some of the code in some places that were obviously just plain wrong and overly complex for what they were doing. Essentially removing `parse_logfile()` which wasn't doing anything useful at all.

**Comment by prologic** [comment]
I'll test locally and hopefully commit and push a fix you can test with

**Comment by prologic** [comment]
I don't think NFS should have any bearing at all IHMO

**Comment by jibanes** [comment]
thanks

**Comment by prologic** [comment]
Hmm it didn't rotate and stopped logging as you described :(

**Comment by prologic** [comment]
I won't know for sure whether this is fixed until Tuesday 14th 00:00 AEST tonight or unless I fake the time and do some unit testing, etc; but hopefully this does fix the issue. I think the filenames weren't generated properly. Please give this a try and let me know how it goes!

**Comment by prologic** [comment]
This still hasn't fixed it (yet) :/

**Comment by prologic** [comment]
Closed via abc04225d0f171dece3748d9a52b2a81213c8291
